### ISSUE TITLE
Protection plugin support

### ISSUE THREAD

**Issue by StargateMC** [LemADEC/WarpDrive#167]
Hi @LemADEC 

I've created what you would need to implement factions support.... I'm working on how to enable this as an 'integration' for your mod.

I already sort out multiple integrations with https://www.spigotmc.org/resources/regenerator.21896/ so if you can help me get started and flesh out an example (and perhaps with Factions, help me finish it) I can move along to the others and do it for you.... Hard bit is splicing in with your mod properly, not the actual code.

Factions code example : http://pastebin.com/MkAiqSPp

EDIT: This would need to be checked both on the destination block position and source.... (can warp out of) and (can warp into).

If any chunks your going into or coming out of deny you, you should be denied.

**Comment by LemADEC** [comment]
That link is invalid, please update it

**Comment by StargateMC** [comment]
http://pastebin.com/raw/MkAiqSPp - or are you refering to the spigotmc link?

**Comment by LemADEC** [comment]
I meant the spigot link

**Comment by StargateMC** [comment]
Ah -  you need to login to spigotmc.org for that one :) Sorry...

**Comment by StargateMC** [comment]
FYI from the page - I have the code available:

My plugin has been tested to work with the following grief prevention plugins;
•Factions 2.8.6+
•Grief Prevention 14+
•Worldguard 6+
•RedProtect 6.5+
•FactionsUUID
•Towny 0.91+
•Landlord 1.3+
•FactionsOne v1.2.2 (v1.8.2 of Factions)

**Comment by StargateMC** [comment]
@LemADEC If you make the WarpDrive mod use a Fake Player with the same name as the pilot when placing / breaking blocks, I can make a custom bukkit plugin to support your mod (take 1 hour tops), rather than the other way round. Might be simpler.

**Comment by LemADEC** [comment]
That's the plan, just not sure when

**Comment by StargateMC** [comment]
Woops. Sorry about the reopen.

Would you be able to do what build craft and mechanism do?

They use [ModName] as the fake player name.

Then until your able to sort out getting the pilot name I can make the plugin provide limited warpdrive support so the warpdrive fake player can warp ships around unclaimed land only (relying on shields etc to defend it).

To be honest that might be the best idea anyway.

**Comment by LemADEC** [comment]
Why would we need a fake player here ?
What's the relation with BuildCraft and Mekanism?

**Comment by StargateMC** [comment]
The quarry from build craft and digital miner from mekanism already use a fake player to mine (break) blocks.

I'm making it so they function in unclaimed land also, but not in claimed. Similar process from an integration Perspective.

**Comment by StargateMC** [comment]
Sorry for the confusion.

Basically the way mods usually integrate with a bukkit based plugin is usually fire BlockPlaceEvent and BlockBreakEvent or a PlayerInteractEvent.

This is the case for the BuildCraft quarry, Mekanism Digital Miner and Thermal Expansion Autonomous activator.

This allows the events to be cancelled by the plugin should they not be able to build or interact there.

See here for an example: https://github.com/BuildCraft/BuildCraft/issues/3323

So long story short, if you were to allow the block placement or break to be cancelled by something else, fail the entire warp with a configurable message and admins and the grief prevention plugins could handle the rest.... if that makes sense?
eg. "Warp failed, unable to place blocks in the destination (or break blocks here), is the territory claimed?" kind of thing..

**Comment by LemADEC** [comment]
I'm a bit confused why simply whitelisting the fake players in the wilderness isn't enough and there would be a need for doing a plugin for special treatment.

**Comment by StargateMC** [comment]
A lot of the plugins will either whitelist them everywhere or not at all. They do this because they know it's useless and don't want to help out hybrid users. Just like forge developers don't like helping them.

Funny considering how many big hybrid servers there are.

**Comment by DraksterAU** [comment]
Example: Massivecraft created an abstract layer for the Bukkit command sender (player) for what seems like 1 purpose: To break cauldron with factions.
It's fixed in 9 characters of code (not lines) but they still have the "Submit a pull request and we will fix it" error displayed when you experience it.
About half a dozen devs, me included have submitted varying pull requests and they've all been denied. Even spawned an offshoot plugin: http://GitHub.com/Markehme/MassiveCoreForgeFixes

**Comment by StargateMC** [comment]
PS most wildernesses are still managed by the plugin.
So they still control builds etc... It is not an unmanaged area.

**Comment by LemADEC** [comment]
Forge has big networks but it's still about 20% of the player base and usually 1 or 2 years late in MC version, while players want the 'cutting' edge snapshot version.
Massive craft is vanilla network, they focus and promote their network, nothing really surprising here.

**Comment by StargateMC** [comment]
Nice!

@lemadec is there any potential bypasses/issues?
I'll test this tonight.

I'll make sure to get perimeter (factions only) to support this tonight... And I'll put this into the forge compatibility plugin I'm making which covers all the plugins supported by Regenerator. It will be ready sometime this weekend :)
The forge compat plugin will also do what perimeter does for factions for the digital miner and quarry.

**Comment by LemADEC** [comment]
I've only tested the collector blocks, so there might be issues with the laser cannon.
It's only using the new [WarpDrive] fake player.

**Comment by StargateMC** [comment]
That's perfect. I'll test the lasers.

Using that is fine, it allows me to let ships warp around in wilderness while protecting claims.

Players can use modded security (cannons, turrets, shields) to protect against griefing on their warp ships.
